Makefile: Add `nonfree` repo to fix model fetching

alex96295 opened the pull request:
* As observed from July 15th, 2025, Infineon requires SSO authentication to download the hyperram model files. Therefore, our current way of fetching the model is broken
* For internal usage, this PR adds support to fetch the model from a cached location, or automatically downloading it, through the variable `CACHED_MODEL`, default to `true`. 
* Open-source users must manually download the models with their credentials.
* `nonfree` reference PR: https://iis-git.ee.ethz.ch/pulp-restricted/hyperbus-nonfree/-/merge_requests/1

**Note:** the flow was successfully tested in-system in chimera: https://iis-git.ee.ethz.ch/github-mirror/chimera/-/pipelines

alex96295 committed:
Makefile: Add `nonfree` repo for model download

As observed from July 15th, 2025, Infineon requires SSO authentication
to donwload the model. For internal usage, we support fetching the model
from a cached location, or automatically downloading it, through the
variable `CACHED_MODEL`, default to `true`. However, open-source users
must manually download the models with their credentials.

phsauter commented:
I think we might want to remove the automatic download if `/etc/iis.version` exists. I see the intention but just because someone is on an iis machine doesn't mean they have access to the repo.

alex96295 commented:
> I think we might want to remove the automatic download if `/etc/iis.version` exists. I see the intention but just because someone is on an iis machine doesn't mean they have access to the repo.

Good point, download will fail anyway if someone does not have access. I guess we can just leave the `make hyperbus-nonfree-init` call and then the command will fail if a IIS person does not have access

alex96295 committed:
Makefile: Do not bound automatic model download to IIS env

We remove automatic download of `nonfree`, that must be done by hand by
systems integratig hyperbus as dependency

alex96295 committed:
Makefile: Allow non-IIS users to point to a cached model location

This is useful if someone wants to leverage this IP in a CI with a
pre-downloaded model. The makefile will just unzip it.

alex96295 commented:
> > I think we might want to remove the automatic download if `/etc/iis.version` exists. I see the intention but just because someone is on an iis machine doesn't mean they have access to the repo.
> 
> Good point, download will fail anyway if someone does not have access. I guess we can just leave the `make hyperbus-nonfree-init` call and then the command will fail if a IIS person does not have access

Errata corrige: we have to remove the whole hunk in `ifneq [...] endif`; however, note that this way systems integrating the hyperbus after this commit will have to manually download the nonfree and add it to the `init` job of their CIs; backward compatibilty suffers slightly, and a major release is required

phsauter committed:
Makefile: Pin merged nonfree model repository
